frappe/deprecation_dumpster.py

```diff
@@ -475,40 +475,61 @@ def tests_timeout(*args, **kwargs):
 	return timeout(*args, **kwargs)
 
 
-@deprecated(
-	"frappe.tests.utils.FrappeTestCase",
-	"2024-20-08",
-	"v17",
-	"use `frappe.tests.UnitTestCase` or `frappe.tests.IntegrationTestCase` respectively",
-)
-def tests_FrappeTestCase(*args, **kwargs):
-	from frappe.tests import IntegrationTestCase
+def get_tests_FrappeTestCase():
+	class CompatFrappeTestCase:
+		def __new__(cls, *args, **kwargs):
+			from frappe.tests import IntegrationTestCase
 
-	return IntegrationTestCase(*args, **kwargs)
+			class _CompatFrappeTestCase(IntegrationTestCase):
+				def __init__(self, *args, **kwargs):
+					deprecation_warning(
+						"2024-20-08",
+						"v17",
+						"Import `frappe.tests.UnitTestCase` or `frappe.tests.IntegrationTestCase` respectively instead of `frappe.tests.utils.FrappeTestCase`",
+					)
+					super().__init__(*args, **kwargs)
 
+			return _CompatFrappeTestCase(*args, **kwargs)
 
-@deprecated(
-	"frappe.tests.utils.IntegrationTestCase",
-	"2024-20-08",
-	"v17",
-	"use `frappe.tests.IntegrationTestCase`",
-)
-def tests_IntegrationTestCase(*args, **kwargs):
-	from frappe.tests import IntegrationTestCase
+	return CompatFrappeTestCase
```

Comment:
Could have been:

```py
class tests_UnitTestCase:
	@deprecated( ... )
	def __new__(cls, *args, **kwargs):
		from frappe.tests import UnitTestCase

		return UnitTestCase.__new__(*args, **kwargs)
```

Reply:
Oh nice, I'll adapt that, the closer the implementation to it's true nature, the better.